Video-Transformers-Java/app/src/main/java/com/tokbox/sample/videotransformers/MainActivity.java

```diff
@@ -0,0 +1,375 @@
+            @Override
+            public void onResponse(Call<GetSessionResponse> call, Response<GetSessionResponse> response) {
+                GetSessionResponse body = response.body();
+                initializeSession(body.apiKey, body.sessionId, body.token);
+            }
+
+            @Override
+            public void onFailure(Call<GetSessionResponse> call, Throwable t) {
+                throw new RuntimeException(t.getMessage());
+            }
+        });
+    }
+
+    private void initializeSession(String apiKey, String sessionId, String token) {
+        Log.i(TAG, "apiKey: " + apiKey);
+        Log.i(TAG, "sessionId: " + sessionId);
+        Log.i(TAG, "token: " + token);
+
+        /*
+        The context used depends on the specific use case, but usually, it is desired for the session to
+        live outside of the Activity e.g: live between activities. For a production applications,
+        it's convenient to use Application context instead of Activity context.
+         */
+        session = new Session.Builder(this, apiKey, sessionId).build();
+        session.setSessionListener(sessionListener);
+        session.connect(token);
+    }
+
+    private void initRetrofit() {
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
+        logging.setLevel(Level.BODY);
+
+        OkHttpClient client = new OkHttpClient.Builder()
+                .addInterceptor(logging)
+                .build();
+
+        retrofit = new Retrofit.Builder()
+                .baseUrl(ServerConfig.CHAT_SERVER_URL)
+                .addConverterFactory(MoshiConverterFactory.create())
+                .client(client)
+                .build();
+
+        apiService = retrofit.create(APIService.class);
+    }
+
+    private void finishWithMessage(String message) {
+        Log.e(TAG, message);
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
+        this.finish();
+    }
+
+    private customVideoTransformer logoTransformer = new customVideoTransformer();
+
+    public class customVideoTransformer implements PublisherKit.CustomVideoTransformer {
+
+        public Bitmap resizeImage(Bitmap image, int width, int height) {
+            Matrix matrix = new Matrix();
+            matrix.postScale(width / (float) image.getWidth(), height / (float) image.getHeight());
+            return Bitmap.createBitmap(image, 0, 0, image.getWidth(), image.getHeight(), matrix, false);
+        }
+
+        @Override
+        public void onTransform(BaseVideoRenderer.Frame frame){
+
+            // Obtain the Y  plane of the video frame
+            ByteBuffer yPlane = frame.getYplane();
+
+            // Get the dimensions of the video frame
+            int videoWidth = frame.getWidth();
+            int videoHeight = frame.getHeight();
+
+            // Get the dimensions of the logo
+            Bitmap image = BitmapFactory.decodeResource(context.getResources(), R.drawable.vonage_logo);
```

Comment:
Does it mean we are going to run this decodeResource for every frame? What's the cost of this operation?

Reply:
BTW I would add that the comment is misleading. This is not exactly getting the dimensions of the logo.

Reply:
we can do this on startup indeed!